src/basic/path-util.c

```diff
@@ -639,8 +639,62 @@ static int check_x_access(const char *path, int *ret_fd) {
         return 0;
 }
 
-int find_executable_full(const char *name, const char *root, bool use_path_envvar, char **ret_filename, int *ret_fd) {
-        int last_error, r;
+static int find_executable_from_path(
+                    const char *name,
+                    const char *root,
+                    const char *element,
+                    char **ret_filename,
+                    int *ret_fd) {
+        int r;
+        int fd = -1;
+        _cleanup_free_ char *dup_element = strdup(element);
+
+        if (dup_element == NULL)
```

Comment:
in our codebase we typically check for points being NULL/non-NULL with C's downgrade-to-bool feature, i.e.:

```c
if (!dup_element)
…
```